### DISCUSSION TITLE
Needs help how to run selenide from web app (jboss + ear)

### DISCUSSION THREAD

**Discussion by Druppov** [selenide/selenide#1938] [Show and tell]
We have a web application (jboss + ear). The application has already implemented a lot of logic tests, now we need to add tests for the UI web application. Tell me please, how this can be implemented? The tests themselves are already configured and tested in the console version (src / test / java). Now you need to transfer them to the web version (src / main / java). But when I first access selenide, I get an exception: 

```java
org.jboss.modules.ModuleClassLoader selenide Method threw java.lang.ClassNotFoundException
```

Maven:

```xml
               < dependency>
			< groupId>com.codeborne</ groupId>
			< artifactId>selenide</ artifactId>
			< version>5.13.1</ version>
		</dependency>

		<dependency>
			<groupId>org.seleniumhq.selenium</groupId>
			<artifactId>client-combined</artifactId>
			<version>3.141.59</version>
		</dependency>

		<dependency>
			<groupId>io.github.bonigarcia</groupId>
			<artifactId>webdrivermanager</artifactId>
			<version>3.8.1</version>
		</dependency>
```

**Comment by asolntsev** [comment]
This is a very strange idea. Why do you need to run tests from a web application? 
Usually people run tests on CI server: Jenkins, Bamboo, Github Actions etc.

**Comment by Druppov** [comment]
We have a web application that implements business logic tests, rest api test and other functions of the main product. Now we decided to add UI tests to the existing web application.

**Comment by asolntsev** [comment]
@Druppov I still don't understand why it should be a web application. Why cannot you just run tests from command line?

**Comment by Druppov** [comment]
We have a web application that reads test data from Google spreadsheets and a tester can run tests through the web environment. Now it is necessary to integrate ui tests into this mechanism.

**Comment by Druppov** [comment]
@asolntsev Is it possible to use selenide in a web application (jboss + ear)? I installed the dependencies, the project builds without errors, but I get error after the first call any selenide object: org.jboss.modules.ModuleClassLoader selenide Method threw java.lang.ClassNotFoundException

**Comment by asolntsev** [comment]
@Druppov Yes, I think it's possible. But you need to find out why JBoss throws `ModuleClassLoader`. I haven't used JBoss for more than 10 years.

NB! I still don't see any reasons why you cannot run from command line tests that read data from Google spreadsheets. Why do you need web application for this? Believe me, I want to help: you are doing it wrong.
